### PR TITLE
ATLauncher: Preliminary work towards pack updating

### PR DESCRIPTION
**Pull request by jamierocks**
Here lies some preparatory work for pack updating for the ATLauncher support. This will be complemented by GH-894, and finished off with a subsequent pull request introducing a tab in the instance settings to expose things :)

---

The commit(s) contained within this pull request have been cherry-picked from my own private fork of MultiMC from circa September 2021.

**For the benefit of PolyMC**: My fork is prior to multiple licences covering the MultiMC codebase, and no longer pulls changes from MultiMC as a result of these developments.

**For the benefit of PolyMC and MultiMC**: I only do development on a PolyMC or MultiMC workspace to resolve merge-conflicts, and to get changes building. No commits are ever cherry-picked onto my fork from a PolyMC or MultiMC workspace - nor between the two in either direction.

### CODE REVIEW

**Comment by txtsd** [comment]
Don't forget to signoff on all your commits using `git commit -s`. It's required now.

**Comment by DioEgizio** [comment]
why did this pr get closed?

File: launcher/modplatform/atlauncher/ATLPackInstallTask.cpp [hunk 1]
```diff
@@ -143,6 +169,116 @@ void PackInstallTask::onDownloadFailed(QString reason)
     emitFailed(reason);
 }
 
+void PackInstallTask::deleteExistingFiles()
+{
+    setStatus(tr("Deleting existing files..."));
+
+    // Setup defaults, as per https://wiki.atlauncher.com/pack-admin/xml/delete
+    VersionDeletes deletes;
+    deletes.folders.append(VersionDelete{ "root", "mods%s%" });
```
Comment (Scrumplex) [inline launcher/modplatform/atlauncher/ATLPackInstallTask.cpp:178]: I think Qt handles `/` just fine, don't think we need to insert the right path separator here

**Review by Scrumplex — APPROVED** [review]
Looks good overall

File: launcher/modplatform/atlauncher/ATLPackInstallTask.cpp (hunk 1 above)
Comment (jamierocks) [reply launcher/modplatform/atlauncher/ATLPackInstallTask.cpp:178]: It does.

They are replaced elsewhere in the code. I've matched the ATLauncher here, as per the above - but also it's the format the API uses, so it's somewhat easier to compare when looking at both.

**Review by flowln — APPROVED** [review]
looks good to me! (sorry for the wait, i was procrastinating :sob:)

some parts of it seem very similar to some of the logic in my PR, so we'll probably end up with some duplicated logic. it's alright though, since AFAIK ATL packs have always been more self-contained than other providers, this shouldn't be that much of an issue :p

**Review by timoreo22 — APPROVED** [review]

**Comment by DioEgizio** [comment]
3 approvals 👀

**Comment by timoreo22** [comment]
Unresolved conversation above that i dunno if i resolve or not

**Pull request merged**